Sources/TuistKit/Simulators/SimulatorRuntimeVersion.swift

```diff
@@ -0,0 +1,111 @@
+import Foundation
+
+struct SimulatorRuntimeVersion: CustomStringConvertible, Equatable, ExpressibleByStringLiteral, Comparable, Decodable {
+    // MARK: - Attributes
+
+    let major: Int
+    let minor: Int?
+    let patch: Int?
+
+    // MARK: - Constructors
+
+    init(major: Int, minor: Int? = nil, patch: Int? = nil) {
+        self.major = major
+        self.minor = minor
+        self.patch = patch
+    }
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        self.init(stringLiteral: try container.decode(String.self))
+    }
+
+    // MARK: - Internal
+
+    func flattened() -> SimulatorRuntimeVersion {
+        return SimulatorRuntimeVersion(major: major,
+                                       minor: minor ?? 0,
+                                       patch: patch ?? 0)
+    }
+
+    // MARK: - CustomStringConvertible
+
+    var description: String {
+        var version = "\(major)"
+        if let minor = minor {
+            version.append(".\(minor)")
+        } else {
+            return version
+        }
+        if let patch = patch {
+            version.append(".\(patch)")
+        } else {
+            return version
+        }
+        return version
+    }
+
+    // MARK: - Equatable
+
+    static func == (lhs: SimulatorRuntimeVersion, rhs: SimulatorRuntimeVersion) -> Bool {
+        return lhs.major == rhs.major &&
+            lhs.minor == rhs.minor &&
+            lhs.patch == rhs.patch
+    }
+
+    // MARK: - Comparable
+
+    static func < (lhs: SimulatorRuntimeVersion, rhs: SimulatorRuntimeVersion) -> Bool {
+        let lhs = lhs.flattened()
+        let rhs = rhs.flattened()
+
+        if lhs.major < rhs.major {
+            return true
+        } else if lhs.major == rhs.major {
+            if lhs.minor! < rhs.minor! {
```

Comment:
Not really because I make those values non-optional by calling [this method](https://github.com/tuist/tuist/pull/746/files#diff-25d19e788788eeb3bdf9eb1835222158R25) before.

Reply:
oh, I was confused because you still needed to force unwrap. Gotcha! 

nit: the `flattend()` function could return the triple `(major, minor, patch)` with no-optionals